### PR TITLE
Remove binlog-server point in time recoveries code & tests

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

This PR removes the legacy code for (Ripple) binlog-server based point in time recoveries. That code is unsupported since v17.

~~TBD:~~ clarify in website docs that the feature is removed (right now it says "legacy", see https://vitess.io/docs/22.0/reference/features/recovery/#point-in-time-recovery-legacy-functionality-based-on-binlog-server)

Docs: https://github.com/vitessio/website/pull/1902. Just completely strip out the legacy docs.

## Related Issue(s)

Closes https://github.com/vitessio/vitess/issues/16673

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [ ] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/test/endtoend/cluster/cluster_util.go [hunk 1]
```diff
@@ -126,15 +126,6 @@ func VerifyRowsInTablet(t *testing.T, vttablet *Vttablet, ksName string, expecte
 	VerifyRowsInTabletForTable(t, vttablet, ksName, expectedRows, "vt_insert_test")
 }
 
-// PanicHandler handles the panic in the testcase.
-func PanicHandler(t testing.TB) {
-	err := recover()
-	if t == nil {
-		return
-	}
```
Comment (mattlord) [inline go/test/endtoend/cluster/cluster_util.go]: If we re-panic here then I think this function and its usage is fine as we'd still always fail the test but hopefully also get some helpful info. Even better IMO to include `debug.Stack()` in the error and log it.

Comment (shlomi-noach) [reply go/test/endtoend/cluster/cluster_util.go]: This hange is merely the resolved conflict caused by https://github.com/vitessio/vitess/pull/17354. It has nothign specific to do with this PR.

**Comment by dbussink** [comment]
Should `test/bin/rippled` also be removed?

**Review by dbussink — APPROVED** [review]

File: test/bin/.keep [hunk 2]
```diff
@@ -0,0 +1 @@
+Do not remove. Ensures existence of directory.
```
Comment (mattlord) [inline test/bin/.keep:1]: Curious why we need the directory?

**Review by mattlord — APPROVED** [review]

File: test/bin/.keep (hunk 2 above)
Comment (shlomi-noach) [reply test/bin/.keep:1]: The `mysql_server_vault` CI job downloads the `vault` binary onto this directory. That CI fails when the directory does not exist.

**Pull request merged**